### DISCUSSION TITLE
Place: how to manage event locations without a full postal address

### DISCUSSION THREAD

**Discussion by saumier** [culturecreates/artsdata-data-model#126] [Data Model]
Signé Laval has several touristic events like the one on the banks of the island of Laval. To enable a good model for these types of events without an address, I think we should add the concept of longitude, latitude to “Place” in Artsdata. And the SHACL will allow for a list of  coordinates as a valid Place even if there is no PostalAddress.  We still need to figure out an event like https://signelaval.com/fr/evenements/15675/rallye-des-rivieres that has 20 points with no starting point.

**Comment by saumier** [comment]
To help with the above case of the event "rallye-des-rivieres", which is an area and not a specific point, I proposed the following addition to Artsdata Places quoted from https://culturecreates.github.io/artsdata-data-model/classes/place.html

> geo property
> https://schema.org/geo - The geo coordinates of the place.
> 
> 
> This property is favoured over adding explicit longitude and latuitude because the property [geo](https://schema.org/geo) not only can include longitude and latitude in class [GeoCoordinates](http://schema.org/GeoCoordinates) but it can also be defined by the class [GeoShape](https://schema.org/GeoShape) which gives flexibility to define parks and areas that do not have a specific starting address.
> 
> This property can satisfy the SHACL validation for Places, removing the absolute requirement of a Place to have a PostalAddress.

**Comment by saumier** [comment]
> ...I think we should add the concept of longitude, latitude to “Place” in Artsdata. And the SHACL will allow for a list of coordinates as a valid Place even if there is no PostalAddress...

When longitude and latitude are used instead of PostalAddress, I propose also including the province and country codes. This will enable searches by PostalAddress to still include events that are in a province or country even when there is no street address or city.  Note that the province and country codes could be inferred from the longitude and latitude when data is imported to Artsdata.

**Comment by saumier** [comment]
The complete proposal

An event's location must be a schema:Place with one of the following shapes:
1. complete schema:PostalAddress
```
 "address": {
    "type": "PostalAddress",
    "addressCountry":"CA",
    "addressLocality": "Gatineau",
    "addressRegion": "QC",
    "postalCode": "J8L 2W1",
    "streetAddress":  "580 Rue Maclaren E"
  }
```
2. schema:longitude, schema:latitude and schema:PostalAddress with province and country
```
"latitude": "45.5873706",
"longitude": "-75.400796",
 "address": {
    "type": "PostalAddress",
    "addressCountry":"CA",
    "addressRegion": "QC"
  }
```
OR
```
 "geo": {
    "@type": "GeoCoordinates",
    "latitude": "40.75",
    "longitude": "-73.98"
  },
 "address": {
    "type": "PostalAddress",
    "addressCountry":"CA",
    "addressRegion": "QC"
  }
```

3. schema:geo with a GeoShape and schema:PostalAddress with province and country

```
 "geo": {
    "@type": "GeoShape",
    ...one of the following: "postalCode", "circle", "polygon", "address (partial)" 
  },
 "address": {
    "type": "PostalAddress",
    "addressCountry":"CA",
    "addressRegion": "QC"
  }
```

**Comment by saumier** [comment]
See also Maude's event CONCERT AU CRÉPUSCULE PAR ÉMILIS about an event on a river https://github.com/culturecreates/tech-support-tickets/issues/81

**Comment by fjjulien** [comment]
An answer is provided in this discussion: https://github.com/culturecreates/artsdata-data-model/discussions/343
